keep/api/core/db.py

```diff
@@ -3248,6 +3248,71 @@ def remove_alerts_to_incident_by_incident_id(
         return deleted
 
 
+class DestinationIncidentNotFound(Exception):
+    pass
+
+
+def merge_incidents_to_id(
+    tenant_id: str,
+    source_incident_ids: List[UUID],
+    destination_incident_id: UUID,
+    merged_by: str | None = None,
+) -> Optional[Incident]:
+    with Session(engine) as session:
+        destination_incident = session.exec(
+            select(Incident)
+            .where(
+                Incident.tenant_id == tenant_id, Incident.id == destination_incident_id
+            )
+            .options(joinedload(Incident.alerts))
+        ).first()
+
+        if not destination_incident:
+            # TODO: maybe allow to create a new incident if the destination incident does not exist
```

Comment:
Sounds valid for me as it is now. Since we provide incident_id as an argument :)